src/lyra/engine/runner.py

```diff
@@ -88,12 +87,16 @@ def variables(self) -> Set[VariableIdentifier]:
                 if isinstance(current, Loop):
                     edges = self.cfg.edges.items()
                     conds = [edge.condition for nodes, edge in edges if nodes[0] == current]
-                    for cond in [c for c in conds if isinstance(c, Call)]:
-                        for arg in [a for a in cond.arguments if isinstance(a, VariableAccess)]:
-                            variable = arg.variable
-                            variables.add(arg.variable)
-                            if isinstance(variable.typ, (StringLyraType, ListLyraType)):
-                                variables.add(LengthIdentifier(variable))
+                    for cond in [c for c in conds if isinstance(c, Call)]:      # TODO: use .ids()?
+                        for arg in cond.arguments:
+                            if isinstance(arg, VariableAccess):
+                                variable = arg.variable
+                                variables.add(arg.variable)
+                                if isinstance(variable.typ, (StringLyraType, ListLyraType)):
+                                    variables.add(LengthIdentifier(variable))
+                            elif isinstance(arg, TupleDisplayAccess):
```

Comment:
Don't we need cases for the other display accesses as well?

Reply:
I wouldn't say so, because you cannot introduce a new variable in the argument of a call, but only on the left of a 'in' and I think you can only have a single variable or a tuple there.